api/agent/drivers/docker/docker_client.go

```diff
@@ -19,6 +19,7 @@ import (
 
 const (
 	retryTimeout = 10 * time.Minute
+	pauseTimeout = 5 * time.Second
```

Comment:
don't think we can avoid this really, but docker likes to fart and we'll hit this at some point for sure. for freezing or unfreezing if this does time out it likely makes the problem worse, since in the freeze case, it will go destroy the container (docker load) potentially requiring making a new one instead of just thawing. in the unfreeze case, we'll just have to launch another which will take longer & more docker load. just tossing this around as a thought experiment... in the unfreeze case specifically it means a call has a handle on the slot and ate a lot of time, which is bad. don't really want 2 phase slots so that we can have triggerThaw->handoutSlot, seems nasty and not sure it buys us much; we could unfreeze 2 and take the first one to unfreeze or potentially something to increase the probability that we can quickly get a thawed container. anyway, not sure it's critical until we start hitting such cases and maybe not worth talking about until we can repro/see some of that behavior.

Reply:
Yeah, I think for now I just want to clamp on that timeout and if it takes more than 5 secs, we are done and we just tear down the container. The timeout is tricky as the longer it is, the longer we expose the container for executing code outside of the allowed thawed state. I'd like to be much more aggressive on it, but at the same time, there's docker/contaierd/runc in the middle, so probably 5 sec is OK for time being...